File: CPP/ByRound/2106/2106_A_Dr_TC.cpp
// Problem: CF 2106 A - Dr. TC
// https://codeforces.com/contest/2106/problem/A

/*
 * Problem: Dr. TC's Binary String Grid
 *
 * Purpose:
 *   This program calculates the total number of '1' bits in a grid formed by
 *   flipping each bit of a given binary string 's' in turn to create n different strings,
 *   and summing all '1' bits across these strings.
 *
 * Algorithm:
 *   - For each test case:
 *     - Read the length n and binary string s.
 *     - For each character in s:
 *       - If it's '1', it contributes (n - 2) + 1 = n - 1 ones to the total
 *         (because in the original string, it's 1; in all other strings, it flips to 0).
 *       - If it's '0', it contributes 1 one to the total (it becomes 1 in only one row).
 *     - Sum up contributions for all characters.
 *
 * Time Complexity: O(n) per test case
 * Space Complexity: O(1) excluding input space
 */

#include <stdio.h>
#include <iostream>
#include <string>

int main() {
  std::ios_base::sync_with_stdio(false); // Synchronize C++ iostream with C stdio for faster input/output
  long t;
  std::cin >> t; // Read number of test cases
  while (t--) {
    long n;
    std::cin >> n; // Read length of binary string
    std::string s;
    std::cin >> s; // Read the binary string
    long cnt(0); // Initialize counter for total number of 1s
    for (size_t p = 0; p < s.size(); p++) {
      cnt += (n - 2) * (s[p] == '1') + 1; // Count contributions from each bit
    }
    printf("%ld\n", cnt); // Print result
  }
}


// https://github.com/VaHiX/codeForces/

File: Java/ByRound/2106/2106_A_Dr_TC.java
// Problem: CF 2106 A - Dr. TC
// https://codeforces.com/contest/2106/problem/A

/*
 * Problem: Dr. TC's Binary String Grid
 * 
 * Purpose:
 *   Given a binary string s of length n, generate n strings where each string is s with one character flipped.
 *   Count the total number of 1s in all generated strings.
 * 
 * Algorithm/Techniques:
 *   - For each string a_i, it is s with the i-th character flipped
 *   - Instead of generating all strings, we can use mathematical approach:
 *     - Let c1 be count of '1's in original string s
 *     - When we flip one character:
 *       - If it was '1', it becomes '0' (decrease by 1)
 *       - If it was '0', it becomes '1' (increase by 1)
 *     - Total number of 1s = (c1-1)*c1 + (c1+1)*(n-c1)
 * 
 * Time Complexity: O(n) per test case
 * Space Complexity: O(1) excluding input storage
 */

import java.util.*;
public class Main{
    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        int t=sc.nextInt();
        while(t--!=0){
            int n=sc.nextInt();
            String s=sc.next();
            int c1=0;
            for(char ch:s.toCharArray()){
                if(ch=='1') c1++; // Count number of 1s in original string
            }
            int total=(c1-1)*c1+(c1+1)*(n-c1); // Mathematical formula to compute total 1s
            System.out.println(total);
        }
    }
}


// https://github.com/VaHiX/CodeForces/